packages/test/test-end-to-end-tests/src/test/loadModes.spec.ts

```diff
@@ -0,0 +1,375 @@
+			return new Promise((resolve) => runtime.once("connected", () => resolve()));
+		}
+	}
+
+	it("Can load a paused container", async () => {
+		const headers: IRequestHeader = {
+			[LoaderHeader.loadMode]: {
+				pauseAfterLoad: true,
+			},
+		};
+		const container2 = await loadContainer(
+			container1.resolvedUrl,
+			testDataObjectFactory,
+			headers,
+		);
+		const dataObject2 = await requestFluidObject<TestDataObject>(container2, "default");
+		const initialValue = dataObject2.value;
+
+		assert.strictEqual(dataObject1.value, dataObject2.value, "counter values should be equal");
+
+		dataObject1.increment();
+		await loaderContainerTracker.ensureSynchronized(container1);
+
+		assert.notEqual(dataObject1.value, dataObject2.value, "counter values should not be equal");
+		assert.notEqual(
+			container1.deltaManager.lastSequenceNumber,
+			container2.deltaManager.lastSequenceNumber,
+			"container sequence numbers should not be equal",
+		);
+		assert.strictEqual(
+			initialValue,
+			dataObject2.value,
+			"sharedCounter2 should still be the initial value",
+		);
+	});
+
+	it("Can load a paused container at a specific sequence number", async () => {
+		// Min 5 ops, max 15 ops
+		const numIncrement = Math.round(Math.random() * 10) + 5;
+		for (let i = 0; i < numIncrement; i++) {
+			dataObject1.increment();
+		}
+		await loaderContainerTracker.ensureSynchronized(container1);
+
+		// Record sequence number we want to pause at, and the expected value at that sequence number
+		const sequenceNumber = container1.deltaManager.lastSequenceNumber;
+		const expectedValue = dataObject1.value;
+
+		const headers: IRequestHeader = {
+			[LoaderHeader.loadMode]: {
+				pauseAfterLoad: true,
+				opsBeforeReturn: "sequenceNumber",
+			},
+			[LoaderHeader.sequenceNumber]: sequenceNumber,
+		};
+		const container2 = await loadContainer(
+			container1.resolvedUrl,
+			testDataObjectFactory,
+			headers,
+		);
+		const dataObject2 = await requestFluidObject<TestDataObject>(container2, "default");
+
+		assert.strictEqual(
+			sequenceNumber,
+			container2.deltaManager.lastSequenceNumber,
+			"container2 should be at the specified sequence number",
+		);
+		assert.strictEqual(
+			expectedValue,
+			dataObject2.value,
+			"sharedCounter2 should still be the expected value",
+		);
+		assert.strictEqual(dataObject1.value, dataObject2.value, "counter values should be equal");
+
+		for (let i = 0; i < numIncrement; i++) {
+			dataObject1.increment();
+		}
+		await loaderContainerTracker.ensureSynchronized(container1);
+
+		assert.notEqual(dataObject1.value, dataObject2.value, "counter values should not be equal");
+		assert.notEqual(
+			container1.deltaManager.lastSequenceNumber,
+			container2.deltaManager.lastSequenceNumber,
+			"container sequence numbers should not be equal",
+		);
+
+		assert.strictEqual(
+			sequenceNumber,
+			container2.deltaManager.lastSequenceNumber,
+			"container2 should still be at the specified sequence number",
+		);
+	});
```

Comment:
Would be good to also include a tests where the latest summary happens to be exactly the sequenceNumber requested.

E.g. do ops up to sequence number 15, force a summary, load a new client paused at 15 and ensure it's happy.